# NPBDetect: A Neural Network Model to Detect Bioactivities from AntiSMASH GBK Files

NPBDetect is a neural network-based tool to predict bioactivities of biosynthetic gene clusters (BGCs) from GBK files generated by AntiSMASH.

NPBDetect is available via:

<table>
<tr>
<td><a href="#cli-installation"><img src="https://edent.github.io/SuperTinyIcons/images/svg/powershell.svg" width="70" title="CLI"></a></td>
<td><a href="#google-colab"><img src="https://edent.github.io/SuperTinyIcons/images/svg/colaboratory.svg" width="70" title="Google Colab"></a></td>
<td><a href="#docker"><img src="https://edent.github.io/SuperTinyIcons/images/svg/docker.svg" width="70" title="Docker"></a></td>
</tr>
</table>

----

## Contents

- [Usage](#usage)
- [CLI Installation](#cli-installation)
- [Google Colab](#google-colab)
- [Docker](#docker)

----

## Usage

1. **Detect biosynthetic gene clusters (BGCs) with antiSMASH 7**

   The first step is to generate `.gbk` files using [antiSMASH 7](https://antismash.secondarymetabolites.org). These GBK files are used by NPBDetect to predict bioactivities.

   - **Online Usage**:
     You can use the antiSMASH web service here:  
     `https://antismash.secondarymetabolites.org`

   - **Local Installation**:
     - Download antiSMASH: `https://antismash.secondarymetabolites.org/#!/download`
     - Follow installation instructions: `http://docs.antismash.secondarymetabolites.org/install/`

   Example command to generate a `.gbk` file:

   ```bash
   antismash <your_input_genome.fasta>
   ```

2. **Use NPBDetect**

   NPBDetect can be run both:
   - Online via Google Colab
   - Locally via CLI (Conda environment) or Docker

----

## CLI Installation

1. **Set up a Conda environment**

   ```bash
   conda create -n npbdetect python=3.10
   conda activate npbdetect
   ```

2. **Install required Python packages**

   ```bash
   pip install pandas
   pip install scikit-learn
   pip install biopython
   pip install torch torchaudio torchvision torchtext torchdata
   ```

3. **Clone the NPBDetect repository**

   ```bash
   git clone git@github.com:dalwindercheema/NPBDetect.git
   cd NPBDetect
   ```

4. **Validate installation and test sample prediction**

   ```bash
   python NPBDetect.py predict \
       -v 1 \
       --gbk test/BGC0000004.gbk \
       --pred HC \
       --out_dir outs/
   ```

----

----

## Google Colab

NPBDetect is also available online through Google Colaboratory.  
No installation required!

ðŸ‘‰ **Launch it here**: [NPBDetect@Google-Colab](https://colab.research.google.com/drive/12zXjqk1DFX8Ouv0rYSAWSoaOheluBys9#scrollTo=MiUQ7HHR2rC8)

----

## Docker

You can also run NPBDetect easily using Docker.

Example command:

```bash
docker run --rm -it \
    --volume <INPUT/OUTPUT_DIR>:/data \
    npbdetect \
    python NPBDetect.py predict \
        -v 1 \
        --gbk /data/BGC0000004.gbk \
        --pred HC \
        --out_dir /data/test1
```

Replace `<INPUT/OUTPUT_DIR>` with the path to your input and output folder.

----

## Citation

If you use NPBDetect in your research, please cite:

> Hemant Goyat, Dalwinder Singh, Sunaina Paliyal, Shrikant Mantri,  
> [**Predicting biological activity from biosynthetic gene clusters using neural networks**](https://www.biorxiv.org/content/10.1101/2024.06.20.599829v1.full.pdf), 2024.

----
